baselines/fedmlb/README.md

```diff
@@ -0,0 +1,627 @@
+python -m fedmlb.dataset_preparation dataset_config.alpha_dirichlet=0.6 
+
+# this will generate the setting for 4. (see above)
+python -m fedmlb.dataset_preparation dataset_config.alpha_dirichlet=0.6 total_clients=500
+```
+Note that, to reproduce those settings, we leverage the `.txt` files
+contained in the`client_data` folder in this project. Such files store
+the specific id of examples in the dataset that are associated with a specific client. 
+For example, the file `client_data/cifar100/balanced/dirichlet0.3_clients100.txt` contains the
+examples for the default setting of this repository. 
+Note that those files are provided by the authors themselves
+on the [paper's GitHub](https://github.com/jinkyu032/FedMLB/tree/main/client_data).
+
+#### Custom config for clients' dataset generation (Tiny-ImageNet)
+
+To generate the partitions for the Tiny-ImageNet settings, i.e.:
+1. Moderate-scale with Dir(0.3), 100 clients, balanced dataset (1000 examples per client).
+2. Large-scale experiments with Dir(0.3), 500 clients, balanced dataset (200 examples per client).
+3. Moderate-scale with Dir(0.6), 100 clients, balanced dataset (1000 examples per client).
+4. Large-scale experiments with Dir(0.6), 500 clients, balanced dataset (200 examples per client).
+
+use the following commands:
+```bash
+# commands to generate clients' dataset partitions with Tiny-imagenet
+# this will generate the setting for 1. (see above)
+python -m fedmlb.dataset_preparation dataset_config.dataset="tiny-imagenet" 
+
+# this will generate the setting for 2. (see above)
+python -m fedmlb.dataset_preparation dataset_config.dataset="tiny-imagenet" total_clients=500
+
+# this will generate the setting for 3. (see above)
+python -m fedmlb.dataset_preparation dataset_config.dataset="tiny-imagenet" dataset_config.alpha_dirichlet=0.6 
+
+# this will generate the setting for 4. (see above)
+python -m fedmlb.dataset_preparation dataset_config.dataset="tiny-imagenet" dataset_config.alpha_dirichlet=0.6 total_clients=500
+```
+> Note: To generate the clients' dataset for the Tiny-Imagenet, the original dataset should be downloaded in advance.\
+> It can be downloaded at http://cs231n.stanford.edu/tiny-imagenet-200.zip. Unzip the folder. \
+> Note: This code supposes to find the folder at the path `/{YOUR_LOCAL_PATH_TO_THE_BASELINE}/FedMLB/tiny-imagenet-200`.
+
+> :warning:
+For Tiny-ImageNet, ensure that the unzipped folder is correctly located at `/{YOUR_LOCAL_PATH_TO_THE_BASELINE}/FedMLB/tiny-imagenet-200`.
+The `tiny-imagenet-200` folder contains three folders (`train`, `val`, `test`) and two `.txt` files.
+
+### Using GPUs
+The code in this repository relies on TF library.
+To make the simulations run on GPUs use the option `client_resources.num_cpus={PER_CLIENT_FRACTION_OF_GPU_MEMORY}`.
+The default is `num_gpus=0.0`. 
+For example, the following command will run on CPU only. 
+```bash
+python -m fedmlb.main # `client_resources.num_gpus=0.0` default
```

Comment:
```suggestion
python -m fedmlb.main # `client_resources.num_gpus=0.0` default

# If for instance you want to allow for two clients running on the same GPU
python -m fedmlb.main client_resources.num_gpus=0.5
```

Reply:
Since this section talks about using gpus, i though it would be nice to show an example command on how to do this. (with it, i'm seeing ~21GB being used, ~8.5GB per client and ~4.4GB for the server). Maybe should we lower it to `num_gpus=1.0`? You decide! Maybe also worth adding a link to the documentation about setting `client_resources`: https://flower.dev/docs/framework/how-to-run-simulations.html#virtualclientengine-resources